cms/static/js/views/import.js

```diff
@@ -2,241 +2,306 @@
  * Course import-related js.
  */
 define(
-    ["domReady", "jquery", "underscore", "gettext"],
-    function(domReady, $, _, gettext) {
+    ["jquery", "underscore", "gettext", "moment", "jquery.cookie"],
+    function($, _, gettext, moment) {
 
         "use strict";
 
-        /********** Private functions ************************************************/
+        /********** Private properties ****************************************/
+
+        var STATE = {
+            'READY'      : 1,
+            'IN_PROGRESS': 2,
+            'SUCCESS'    : 3,
+            'ERROR'      : 4
+        }
+
+        var current = { stage: 0, state: STATE.READY };
```

Comment:
@AkA84 nit: since you've started improving the code significantly, it would be great to also provide named constants for stages, e.g. 

    var STAGE = {
        UPLOADING: 1,
        UNPACKING: 2,
        ...
    };

In such case it would even allow keeping `<=` and `>=` comparisons.